tests/lib/external/snapd-testing-tools/tests/spread-filter/task.yaml

```diff
@@ -0,0 +1,93 @@
+
+    # Scenario: 2 unit tests changed
+    spread-filter -r config.yaml -s ubuntu -b google -c source/lib1/lib1_test.go -c source/lib2/lib2_test.go > res
+    MATCH "^google:ubuntu:tests/unit/c-unit-tests-clang google:ubuntu:tests/unit/c-unit-tests-gcc google:ubuntu:tests/unit/go$" < res
+
+    # Scenario: 1 unit tests and 1 test changed
+    spread-filter -r config.yaml -s ubuntu -b google -c source/lib1/lib1_test.go -c tests/main/test1/noexist.go > res
+    MATCH "^google:ubuntu:tests/unit/c-unit-tests-clang google:ubuntu:tests/unit/c-unit-tests-gcc google:ubuntu:tests/main/test1 google:ubuntu:tests/unit/go$" < res
+
+    # Scenario: a doc file changed
+    spread-filter -r config.yaml -s ubuntu -b google -c README.md > res
+    test "$(wc -w < res)" -eq 0
+
+    # Scenario: a doc file which does not exist changed (deleted)
+    spread-filter -r config.yaml -s ubuntu -b google -c DELETED.md > res
+    test "$(wc -w < res)" -eq 0
+
+    # Scenario: a doc file and a the nested library changed
+    spread-filter -r config.yaml -s ubuntu -b google -c README.md -c tests/lib/nested.sh > res
+    MATCH "^google:ubuntu:tests/nested/$" < res
+    
+    # Scenario: 1 test and also another which is substring of previous one changed
+    spread-filter -r config.yaml -s ubuntu -b google -c README.md -c tests/main/test1/task.yaml -c tests/main/test1b/task.yaml > res
+    MATCH "^google:ubuntu:tests/main/test1b google:ubuntu:tests/main/test1$" < res
+
+    # Scenario: a file in the test suite but not in a test dir changed
+    spread-filter -r config.yaml -s ubuntu -b google -c tests/core/other/lib1 > res
+    test "$(wc -w < res)" -eq 0
+
+    # Scenario: the nested lib and a nested test are changed (check the test is cleaned)
+    spread-filter -r config.yaml -s ubuntu -b google -c tests/lib/nested.sh -c tests/nested/test1/task.yaml > res
+    MATCH "^google:ubuntu:tests/nested/$" < res
+
+    # Scenario: the nested lib, a source file and a nested test are changed (check just tests/ survives)
+    spread-filter -r config.yaml -s ubuntu -b google -c tests/lib/nested.sh -c source/lib1/lib1 -c tests/nested/test1/task.yaml > res
+    MATCH "^google:ubuntu:tests/$" < res
+
+    # Scenario: check -v parameter works
+    spread-filter -r config.yaml -s ubuntu -b google -c tests/lib/nested.sh -v > res
+    MATCH "cleaning executions" < res
+
+    # Scenario: check mandatory parameters
+    ARG_TEXT="spread-filter: error: the following arguments are required"
+    spread-filter -r config.yaml -b google -c tests/lib/nested.sh 2>&1 | MATCH "$ARG_TEXT"
+    spread-filter -r config.yaml -s ubuntu -c tests/lib/nested.sh 2>&1 | MATCH "$ARG_TEXT"
+    spread-filter -s ubuntu -b google -c tests/lib/nested.sh 2>&1 | MATCH "$ARG_TEXT"
+
+    # Scenario: check the config rules file
+    FILE_TEXT="spread-filter: rules file 'noexist.yaml' does not exist"
+    spread-filter -r noexist.yaml -s ubuntu -b google -c tests/lib/nested.sh 2>&1 | MATCH "$FILE_TEXT"
+  
```

Comment:
Remove extra line

Reply:
I see all the tests having an extra line at the end